### ISSUE TITLE
Make xpass a failure again 

### ISSUE THREAD

**Issue by RonnyPfannschmidt** [pytest-dev/pytest#11467]
#### What's the problem this feature will solve?

Some years ago we accepted a change which made xpass not a failure

I recently hit multiple cases where it was missed that a test was fixed

That's not an acceptable default

**Comment by nicoddemus** [comment]
This means changing the default of the `strict` parameter to `True`?

**Comment by RonnyPfannschmidt** [comment]
That's would be the starting point, In the long run it should go away

**Comment by TanyaAgarwal28** [comment]
Hi, How we can test this scenario, I can't see xpass as a marker in pytest.

**Comment by RonnyPfannschmidt** [comment]
@TanyaAgarwal28 xpass is a test outcome for when a xfail marked test passed "unexpectedly"

**Comment by TanyaAgarwal28** [comment]
Yeah Got it so what we are facing issue here?

**Comment by RonnyPfannschmidt** [comment]
@TanyaAgarwal28 the problem is that its a misfeature - it hides  important change for a non-flaky testsuite - fixing a previously failing test by accident 

when non-strict xfail was introduced, it was a major mistake not to use a new mark

**Comment by TanyaAgarwal28** [comment]
So basically we have to revert this change and if xfail testcase is by mistakly fixed then also in outcome it should be xfail??

**Comment by TanyaAgarwal28** [comment]
Overview of 4 scenarios for xfail and strict_xfail
xfail testcase which is expectedly fail:
import pytest

@pytest.mark.xfail
def test_example_expected_fail():
    assert 3 + 3 == 7  # This test is marked as expected fail
 
![image](https://github.com/pytest-dev/pytest/assets/67827730/52b9e8f4-bad0-4c45-b7ba-c23edd180d8d)

xfail testcase which is passed:
import pytest

@pytest.mark.xfail
def test_example_expected_fail():
    assert 3 + 3 == 6  # This test is marked as expected fail

![image](https://github.com/pytest-dev/pytest/assets/67827730/ae8a20ab-b42f-4779-8a10-a2affcaa83cd)

non_strict_xfail testcase is passed:
import pytest

@pytest.mark.non_strict_xfail
def test_example_expected_fail():
    assert 3 + 3 == 6 # This test is marked as expected fail
![image](https://github.com/pytest-dev/pytest/assets/67827730/8f23c4cc-c7e7-4474-a52b-8706e7accccd)

**Comment by nicoddemus** [comment]
> So basically we have to revert this change and if xfail testcase is by mistakly fixed then the outcome should be passed??
And for non_strict_xfail we should have new mark??

IMHO what should be done:

1. `pytest.mark.xfail` has a `strict` parameter, which defaults to `False` today. We want to issue a warning that this default will change to `True` in the future, but only if the `strict` flag has not been passed explicitly.
2. A few major releases later, we change the default to `True` and remove the warning.

Is this what you have in mind @RonnyPfannschmidt ?

**Comment by RonnyPfannschmidt** [comment]
Correct, once the default is switched I'd like to ensure xpass is either a warning or a failure in any case

**Comment by TanyaAgarwal28** [comment]
okay Got it
Thanks

**Comment by bluetech** [comment]
We now have strict mode which enables `strict_xfail`.  Maybe that's enough to close this issue?

**Comment by RonnyPfannschmidt** [comment]
this issue is about making it default to true
